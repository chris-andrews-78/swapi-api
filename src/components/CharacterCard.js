import React, {Component} from 'react';
import {SwapiContext} from "../context/SwapiAPI";

export default class CharacterCard extends Component {
    static contextType = SwapiContext;
    render(){
      const { isCharacterFound, selectedCharacter } = this.context;
      return (
        <div>
          {(selectedCharacter) ? (
            <div className="sw-app-card">
              {(isCharacterFound) ? (
              <div>
                <div className="sw-app-card__header">
                    <img src={"images/img-avatar-" + selectedCharacter.name + ".png"} alt={selectedCharacter.name} onError={(e)=>{e.target.onerror = null; e.target.src="images/img-avatar-default.png"}} className="sw-app-avatar sw-app-avatard--card" /> 
                    <span>{selectedCharacter.name}</span>
                </div>
                <div className="sw-app-card__body">
                  <div className="sw-app-card__row">
                    <span><strong>Gender</strong></span> <span>{selectedCharacter.gender}</span>
                  </div>
                  <div className="sw-app-card__row">
                    <span><strong>Birth year</strong></span> <span>{selectedCharacter.birth_year}</span>
                  </div>
                  <div className="sw-app-card__row">
                    
                    <span><strong>Height</strong></span> <span>{selectedCharacter.height}</span>
                  </div>
                  <div className="sw-app-card__row">
                    <span><strong>Mass</strong></span> <span>{selectedCharacter.mass}</span>
                  </div>
                  <div className="sw-app-card__row">
                    <span><strong>Hair Colour</strong></span> <span>{selectedCharacter.hair_color}</span>
                  </div>
                </div>
              </div>
              ) : <p>Sorry no character found, please try again</p>}
            </div>
          ) : ""}
        </div>
      );
    }
}