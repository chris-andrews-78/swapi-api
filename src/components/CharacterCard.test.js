import React from 'react';
import renderer from 'react-test-renderer';
import CharacterCard from './CharacterCard';
import SwapiProvider from "../context/SwapiAPI";

test('Test Character Card', () => {

    const component = renderer.create(
        <CharacterCard /> //Need to figure out how to pass context to tests
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});