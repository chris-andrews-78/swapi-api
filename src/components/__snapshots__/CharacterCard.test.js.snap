// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Character Card 1`] = `
<div>
  
</div>
`;
